AutoCollection/HttpRequests.ts

```diff
@@ -72,13 +72,17 @@ class AutoCollectHttpRequests {
     private _initialize() {
         this._isInitialized = true;
 
-        var originalHttpServer = http.createServer;
-        http.createServer = (onRequest) => {
-            // todo: get a pointer to the server so the IP address can be read from server.address
-            return originalHttpServer((request:http.ServerRequest, response:http.ServerResponse) => {
+        const wrapOnRequestHandler: Function = (onRequest?: Function) => {
+            if (!onRequest) {
+                return undefined;
+            }
+            if (typeof onRequest !== 'function') {
```

Comment:
This is different behavior than the non-wrapped function. In node 6.11 at least, I still get an object returned for `http.createServer(null, "abc")`

Reply:
Same for `http.createServer(null, undefined)`

Reply:
To be honest, I do not fully get your point. The function `http.createServer` does not accept a second parameter (or at least is not using it). Only the `https` version is. And the function is [always returning a server object](https://github.com/nodejs/node/blob/v6.11.2/lib/http.js#L43-L45), no matter what you pass to it. Digging deeper and having a look at the Server constructor, it will only attach the listener function if it is truthy: https://github.com/nodejs/node/blob/v6.11.2/lib/_http_server.js#L233-L235 and then in the EventEmitter's addListener, it will [throw an exception when the listener is not a function](https://github.com/nodejs/node/blob/v6.11.2/lib/events.js#L215-L216).

So I think the patch mimics the behavior of the unpatched version.

Actually the current appinsights' master is having a different behavior I think. It always passes a valid function to `createServer` even though the user-provided argument might not be one: https://github.com/Microsoft/ApplicationInsights-node.js/blob/0131e35d29c268c8c99368a7a3a6435828186569/AutoCollection/HttpRequests.ts#L78-L91

Reply:
While digging through the Node.js code, I found that just patching 'EventEmitter.on' isn't enough, because we also have to wrap 'addListener' (https://github.com/nodejs/node/blob/v6.11.2/lib/events.js#L275-L279). And since the createServer is calling addListener eventually anyways, we do not have to patch that. Will update the PR accordingly.

Reply:
> Actually the current appinsights' master is having a different behavior I think. It always passes a valid function to createServer even though the user-provided argument might not be one

Great catch!

My point before was over `if (typeof onRequest !== 'function')` in `wrapOnRequestHandler` potentially throwing when it wasn't expected, but I missed the truthiness check right above it that returns `undefined`, avoiding this problem :smile:

Reply:
Yeah, hope it covers all cases. At least it should for the current implementation of node.

@OsvaldoRosado i fixed the `.addListener()` and `.on()` patching and added a comment explaining why it was done the way it is: https://github.com/0x7f/ApplicationInsights-node.js/blob/b069fc2e5ba99ec95240187fe645535ef9192513/AutoCollection/HttpRequests.ts#L100-L126

wdyt?

Reply:
LGTM. Thanks for this!